lib/events/dynamoevents/dynamoevents.go

```diff
@@ -499,20 +502,29 @@ func (l *Log) SearchEvents(fromUTC, toUTC time.Time, filter string, limit int) (
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	input := dynamodb.QueryInput{
-		KeyConditionExpression:    aws.String(query),
-		TableName:                 aws.String(l.Tablename),
-		ExpressionAttributeValues: attributeValues,
-		IndexName:                 aws.String(indexTimeSearch),
-	}
 	start := time.Now()
-	out, err := l.svc.Query(&input)
-	if err != nil {
-		return nil, trace.Wrap(err)
+	for {
```

Comment:
Lets use some memory limit and not the number of iterations. DynamoDB pagination is based on the size of query result (max 900kb) meaning that  900kb may have just a few events in theory. So 300 iteration potentially can use 300mb of RAM per request. @r0mant @awly is there a way to determine the RAM available and take % from it as a max buffer size during iteration? (this does not sound as reliable solution of course but what are other options until we add streaming?)



Since we do not have streaming 



We have a hard limit of 10k events (the limit value). 

DynamoDB pagination is based on size (not the number of events), so if size is too greate it would consume all memory regardless of this

Reply:
@alex-kovoy having dynamic memory computation is overly complex, let's not do that. I'm OK to set some large hardcoded value of 100MB per result set though instead of the number of iterations, this is a good idea.

Reply:
We can add some queue and place incoming requests on hold while the queue is being processed. Pending requests will have some timeout values.

Reply:
No need for queue and processing of incoming requests. If we want to make it right, we should expose iterator to the higher level API to allow for efficient pagination.

Reply:
When hit with several requests per second, this 100mb limit may lead to DoS.

Reply:
I might also add that a potentially unbounded loop should make progress. The primary key (`lastEvaluatedKey`) seems a likely candidate for progression which should change between two sequential invocations unless it is empty.